### PR TITLE
chore: added router specific destinations latency metric

### PR DESCRIPTION
**Pull request by anantjain45823**
## Description of the change

> Adding latency metric for router transform destinations like MP.
## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/controllers/destination.ts [hunk 1]
```diff
@@ -138,6 +140,12 @@ export default class DestinationController {
       'Native(Router-Transform):: Response from transformer::',
       JSON.stringify(ctx.body),
     );
+    stats.timing('dest_transform_request_latency', startTime, {
+      destination,
+      version: 'v0',
+      type: 'router',
```
Comment (sanpj2292) [inline src/controllers/destination.ts:146]: Instead of `type` let's put it as `feature`(common convention)

Comment (anantjain45823) [reply src/controllers/destination.ts:146]: OKay

File: src/controllers/destination.ts [hunk 2]
```diff
@@ -179,6 +188,11 @@ export default class DestinationController {
       'Native(Process-Transform-Batch):: Response from transformer::',
       JSON.stringify(ctx.body),
     );
+    stats.timing('dest_transform_request_latency', startTime, {
+      destination,
+      feature: 'batch',
+      version:"v0",
+    });
```
Comment (anantjain45823) [inline src/controllers/destination.ts:195]: Have not incorporated metatags as we don't have them.

File: src/controllers/destination.ts (hunk 1 above)
Comment (anantjain45823) [reply src/controllers/destination.ts]: Done

**Review by mihir-4116 — APPROVED** [review]
LGTM

**Review by sanpj2292 — APPROVED** [review]

**Review by mihir-4116 — APPROVED** [review]

**Pull request merged**